### ISSUE TITLE
DPF: Test and Sample Updates

### ISSUE THREAD

**Issue by zeier** [agera-edc/DataSpaceConnector#161]
Feature #20 
After #115 

Upstream https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/issues/902

## Description

Samples and tests currently use the `InlineDataFlowController`, which was a temporary solution until the Data Plane Framework was in place.

Upstream issue https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/issues/871 will use the `HttpDataFlowController` to perform communication between the Control Plane and the Data Plane.

Test and samples need to be updated to use the DPF instead of the `InlineDataFlowController`.

Integration tests and implementation of missing pieces are part of #173. 

## Acceptance Criteria

- [x] Usage of `InlineDataflowController` is replaced by the `HTTPDataFlowController`
- [x] Samples and tests leverage the Data Plane for data transfer

## Tasks

- [x] Review PR for upstream issue 871
- [x] Update samples and integration tests

**Comment by zeier** [comment]
Put on hold, conversation on Tuesday 22/03 required to move forward.

**Comment by cpeeyush** [comment]
Observability will be taken care by #173

**Comment by cpeeyush** [comment]
Upstream PR https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/pull/1060
